CHANGELOG.md

```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Parameter `meta: dict` on `BaseEvent`.
 - `AzureOpenAiTextToSpeechDriver`.
+- `JsonSchemaRule` for instructing the LLM to output a JSON object that conforms to a schema.
```

Comment:
Responding with an inline comment, so it's easier to reply.

> Question: should this be the mechanism by which we enable features like OpenAI Structured Outputs? If so, we'll need to make the prompt rendering happen at the Driver level rather than the Task level.

If we didn't use this mechanism for OpenAI Structured Outputs, then what would the mechanism for that look like? Would it require passing a json schema? (I would expect so) If it requires passing a json schema, then we'd have two choices, which doesn't seem like the right approach (imagine that we had taken that approach for tools 🤮 ).

Seems like making prompt rendering at the Driver level is the way forward because I don't really see duplicate ways of passing in a json schema as a viable option. This could technically wait until introducing native structured outputs, right?

(Also, if we do use this rule for native structure outputs, what mechanism will we use to opt-in/out of structure outputs when a driver supports it? Should we that the same approach as a `use_native_tools` flag?)

Reply:
> Would it require passing a json schema?
Yes, it'd likely require passing a schema to the Prompt Driver 🤢. 

Rendering at the Prompt Driver level allows us to lean on more provider-specific nuances so I think it's a worthwhile change. But as you point out it's not required for this PR -- this can still serve as a good intermediate solution.